### PR TITLE
SNA

### PR DESCRIPTION
**Pull request by dev-otpless-tech**

### CODE REVIEW

File: Example/OtplessSDK/Info.plist [hunk 1]
```diff
@@ -61,5 +61,22 @@
 		<string>UIInterfaceOrientationPortrait</string>
 		<string>UIInterfaceOrientationLandscapeLeft</string>
 	</array>
+    <key>NSAppTransportSecurity</key>
+        <dict>
+            <key>NSAllowsArbitraryLoads</key>
+            <true/>
+            <key>NSExceptionDomains</key>
+            <dict>
+                <key>80.in.safr.sekuramobile.com</key>
```
Comment (dev-otpless-tech) [inline Example/OtplessSDK/Info.plist:70]: add jio url as well

File: OtplessSDK/Classes/CellularConnectionManager.swift [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+//
+
+import Network
+import os
+
+typealias ResultHandler = (ConnectionResult) -> Void
+
+/// Force connectivity to cellular only
+
+@available(iOS 12.0, *)
```
Comment (dev-otpless-tech) [inline OtplessSDK/Classes/CellularConnectionManager.swift:15]: remove

File: OtplessSDK/Classes/CellularConnectionManager.swift [hunk 3]
```diff
@@ -0,0 +1,500 @@
...
+            if let redirectURL =  URL(string: String(cleanRedirect)) {
+                return RedirectResult(url: redirectURL.host == nil ? URL(string: redirectURL.description, relativeTo: requestUrl)! : redirectURL, cookies: nil)
+            } else {
+                return nil
+            }
+        }
+        return nil
+    }
+    
+//    func parseCookies(url: URL, response: String, existingCookies: [HTTPCookie]?) -> [HTTPCookie]? {
```
Comment (dev-otpless-tech) [inline OtplessSDK/Classes/CellularConnectionManager.swift:261]: remove this

File: OtplessSDK/Classes/NativeWebBridge.swift [hunk 4]
```diff
@@ -205,7 +205,30 @@ class NativeWebBridge {
...
+                                    let script = tempScript.replacingOccurrences(of: "\n", with: "")
+                                    self.callJs(webview: self.webView, script: script)
+                                }
+                            } catch {
+                                
+                            }
+                        }
+                    )
+                } else {
+                    // handle case when unable to create URL from string
```
Comment (dev-otpless-tech) [inline OtplessSDK/Classes/NativeWebBridge.swift:229]: handle this case

File: OtplessSDK/Classes/NativeWebBridge.swift [hunk 5]
```diff
@@ -205,7 +205,30 @@ class NativeWebBridge {
...
+                        completion: { silentAuthResponse in
+                            do {
+                                let jsonData = try JSONSerialization.data(withJSONObject: silentAuthResponse, options: .prettyPrinted)
+                                if let jsonStr = String(data: jsonData, encoding: .utf8) as String? {
+                                    let tempScript = "onCellularNetworkResult(" + jsonStr + ")"
+                                    let script = tempScript.replacingOccurrences(of: "\n", with: "")
+                                    self.callJs(webview: self.webView, script: script)
+                                }
+                            } catch {
+
```
Comment (dev-otpless-tech) [inline OtplessSDK/Classes/NativeWebBridge.swift:224]: handle this case

File: OtplessSDK/Classes/NativeWebBridge.swift [hunk 6]
```diff
@@ -271,3 +294,12 @@ public protocol BridgeDelegate: AnyObject {
     func hideLoader()
     func dismissView()
 }
+
+extension NativeWebBridge {
+    func forceOpenURLOverMobileNetwork(url: URL, completion: @escaping ([String: Any]) -> Void) {
+        if #available(iOS 12.0, *) {
```
Comment (dev-otpless-tech) [inline OtplessSDK/Classes/NativeWebBridge.swift:300]: remove the check

**Review by sparsh-otpless — APPROVED** [review]

**Pull request merged**